internal/transport/transport_test.go

```diff
@@ -2243,3 +2244,27 @@ func TestTCPUserTimeout(t *testing.T) {
 		}
 	}
 }
+
+// TestRequestInfoFoundInStreamContext tests whether data passed in as callheader data gets
+// propagated into the context object as a credentials.RequestInfo object.
+func TestRequestInfoFoundInStreamContext(t *testing.T) {
+	serverConfig := &ServerConfig{}
+	server, client, cancel := setUpWithOptions(t, 0, serverConfig, suspended, ConnectOptions{})
+	defer cancel()
+	defer server.stop()
+	defer client.Close()
+	ch := &CallHdr{
+		Method: "someService/Method",
+	}
+	s, err := client.NewStream(context.Background(), ch)
```

Comment:
Please add a deadline to all contexts:
```go
ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
defer cancel()
...
```

To prevent tests from hanging for many minutes before the overall timeout expires.  (Yes, many of our existing tests don't do this.)